In [44]:
import numpy as np

In [15]:
import pandas as pd
df = pd.read_csv('/Users/krishnasoni/Downloads/cc.csv')
df.head()

Age  Gender  Income            Education Marital Status  \
0   25  Female   50000    Bachelor's Degree         Single   
1   30    Male  100000      Master's Degree        Married   
2   35  Female   75000            Doctorate        Married   
3   40    Male  125000  High School Diploma         Single   
4   45  Female  100000    Bachelor's Degree        Married   

   Number of Children Home Ownership Credit Score  
0                   0         Rented         High  
1                   2          Owned         High  
2                   1          Owned         High  
3                   0          Owned         High  
4                   3          Owned         High

In [16]:
df.info

<bound method DataFrame.info of      Age  Gender  Income            Education Marital Status  \
0     25  Female   50000    Bachelor's Degree         Single   
1     30    Male  100000      Master's Degree        Married   
2     35  Female   75000            Doctorate        Married   
3     40    Male  125000  High School Diploma         Single   
4     45  Female  100000    Bachelor's Degree        Married   
..   ...     ...     ...                  ...            ...   
159   29  Female   27500  High School Diploma         Single   
160   34    Male   47500   Associate's Degree         Single   
161   39  Female   62500    Bachelor's Degree        Married   
162   44    Male   87500      Master's Degree         Single   
163   49  Female   77500            Doctorate        Married   

     Number of Children Home Ownership Credit Score  
0                     0         Rented         High  
1                     2          Owned         High  
2                     1          Owne

In [17]:
df.shape

(164, 8)

In [18]:
df.columns

Index(['Age', 'Gender', 'Income', 'Education', 'Marital Status',
       'Number of Children', 'Home Ownership', 'Credit Score'],
      dtype='object')

In [20]:
df.isnull().sum()

Age                   0
Gender                0
Income                0
Education             0
Marital Status        0
Number of Children    0
Home Ownership        0
Credit Score          0
dtype: int64

In [21]:
df.describe()

Age         Income  Number of Children
count  164.000000     164.000000          164.000000
mean    37.975610   83765.243902            0.652439
std      8.477289   32457.306728            0.883346
min     25.000000   25000.000000            0.000000
25%     30.750000   57500.000000            0.000000
50%     37.000000   83750.000000            0.000000
75%     45.000000  105000.000000            1.000000
max     53.000000  162500.000000            3.000000

In [22]:
df.dtypes

Age                    int64
Gender                object
Income                 int64
Education             object
Marital Status        object
Number of Children     int64
Home Ownership        object
Credit Score          object
dtype: object

In [29]:
df["Credit Score"].value_counts()
df["Credit Score"].value_counts(normalize=True)


Series([], Name: proportion, dtype: float64)

In [34]:
import matplotlib.pyplot as plt
import seaborn as sns

sns.set_style("whitegrid")

Dataset contains 164 records/
Target variable slightly balanced/
Income is right-skewed/
Credit Score is best for pridiction/
There is no empty row